crates/cadenza-eval/src/eval.rs

```diff
@@ -21,6 +21,52 @@ use cadenza_syntax::{
     span::Span,
 };
 
+/// Helper function to intern a string from rowan's SyntaxText efficiently.
+///
+/// This function optimizes the interning process for the common case where
+/// text is stored as a single contiguous chunk (e.g., simple identifiers).
+///
+/// Previously we used `.to_string().as_str().into()` which allocates a String
+/// then immediately copies it into the interner. Now we optimize for single
+/// chunks while still handling multi-chunk text correctly.
+#[inline]
+fn intern_syntax_text(text: &rowan::SyntaxText) -> InternedString {
+    // Track state during iteration to handle single vs multi-chunk efficiently
+    enum State {
+        Empty,
+        Single(String),
+        Multi(String),
+    }
+
+    let final_state: std::result::Result<State, std::convert::Infallible> =
+        text.try_fold_chunks(State::Empty, |state, chunk| {
+            Ok(match state {
+                State::Empty => {
+                    // First chunk
+                    State::Single(chunk.to_owned())
```

Comment:
We're still allocating here so we just moved the problem elsewhere. I think the better approach would be having the interning be aware of the SyntaxText type. With that, we should be able to avoid calling to_string and/or to_owned